DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,305 @@
+# Dev Environment Configuration
```

Comment:
Since this is specific for Linux I would change the title to "Linux Dev Environment Configuration" and I would do the same thing for the file name